python/hail/methods/statgen.py

```diff
@@ -1847,18 +1847,161 @@ def rrm(call_expr):
+    population comprising :math:`K` homogeneous modern populations that have
+    each diverged from a single ancestral population (a `star phylogeny`). Each
+    sample is assigned a population by sampling from the categorical
+    distribution :math:`\pi`. Note that the actual size of each population is
+    random.
+
+    Variants are modeled as bi-allelic and unlinked. Ancestral allele
+    frequencies are drawn independently for each variant from a frequency
+    spectrum :math:`P_0`. The extent of genetic drift of each modern population
+    from the ancestral population is defined by the corresponding :math:`F_{ST}`
+    parameter :math:`F_k` (here and below, lowercase indices run over a range
+    bounded by the corresponding uppercase parameter, e.g. :math:`k = 1, \ldots,
+    K`). For each variant and population, allele frequencies are drawn from a
+    `beta distribution <https://en.wikipedia.org/wiki/Beta_distribution>`__
+    whose parameters are determined by the ancestral allele frequency and
+    :math:`F_{ST}` parameter. The beta distribution gives a continuous
+    approximation of the effect of genetic drift. We denote sample population
+    assignments by :math:`k_n`, ancestral allele frequencies by :math:`p_m`,
+    population allele frequencies by :math:`p_{k, m}`, and diploid, unphased
+    genotype calls by :math:`g_{n, m}` (0, 1, and 2 correspond to homozygous
+    reference, heterozygous, and homozygous variant, respectively).
+    
+    The generative model is then given by:
+
+    .. math::
+        k_n \,&\sim\, \pi
+
+        p_m \,&\sim\, P_0
+
+        p_{k,m} \mid p_m\,&\sim\, \mathrm{Beta}(\mu = p_m,\, \sigma^2 = F_k p_m (1 - p_m))
+
+        g_{n,m} \mid k_n, p_{k, m} \,&\sim\, \mathrm{Binomial}(2, p_{k_n, m})
+
+    The beta distribution by its mean and variance above; the usual parameters
+    are :math:`a = (1 - p) \frac{1 - F}{F}` and :math:`b = p \frac{1 - F}{F}` with
+    :math:`F = F_k` and :math:`p = p_m`.
+
+    The resulting dataset has the following fields.
+
+    Global fields:
+
+    - **num_populations** (*Int32*) -- Number of populations.
+    - **num_samples** (*Int32*) -- Number of samples.
+    - **num_variants** (*Int32*) -- Number of variants.
+    - **pop_dist** (*Array[Float64]*) -- Population distribution indexed by
+      population.
+    - **fst** (*Array[Float64]*) -- :math:`F_{ST}` values indexed by
+      population.
+    - **ancestral_af_dist** (*Struct*) -- Description of the ancestral allele
+      frequency distribution.
+    - **seed** (*Int32*) -- Random seed.
+
+    Row fields:
+
+    - **v** (:class:`.TVariant`) -- Variant (key field).
+    - **ancestral_af** (*Float64*) -- Ancestral allele frequency.
+    - **af** (*Array[Float64]*) -- Modern allele frequencies indexed by
+      population.
+
+    Column fields:
+
+    - **s** (*String*) - Sample ID (key field).
+    - **pop** (*Int32*) -- Population of sample.
```

Comment:
Can you make all of these parameter types link to the Hail Types? For Array[Float64], you can do :class:TArray of :class:TFloat64.

Also, assuming the format for import_vcf is the one we are going with, can you make the formatting consistent with that. For example

```
- `v` (:class:`.TVariant`) -- variant
```